### PR TITLE
container: ViewDB: cleanup error-types

### PR DESCRIPTION
**Pull request by thaJeztah**
- relates to https://github.com/moby/moby/pull/43624

some changes I had in a local branch while I was reviewing https://github.com/moby/moby/pull/43624 - removes some custom error-types that didn't have to be exported, and fixes some errors to prevent producing 500 "internal server errors".

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by thaJeztah** [comment]
One failure; looks like it may be related

```
=== RUN   TestDockerSuite/TestLinksMultipleWithSameName
    docker_cli_links_test.go:251: assertion failed: 
        Command:  /usr/local/cli/docker run --link upstream-a:upstream --link upstream-b:upstream busybox sh -c ping -c 1 upstream
        ExitCode: 125
        Error:    exit status 125
        Stdout:   
        Stderr:   /usr/local/cli/docker: Error response from daemon: name is reserved.
        See '/usr/local/cli/docker run --help'.
        
        
        Failures:
        ExitCode was 125 expected 0
        Expected no error
    --- FAIL: TestDockerSuite/TestLinksMultipleWithSameName (1.00s)
```

File: container/view.go [hunk 1]
```diff
@@ -312,7 +285,7 @@ func (v *memdbView) getNames(containerID string) []string {
 func (v *memdbView) GetID(name string) (string, error) {
 	s, err := v.txn.First(memdbNamesTable, memdbIDIndex, name)
 	if err != nil {
-		return "", err
+		return "", errdefs.System(err)
 	}
 	if s == nil {
 		return "", ErrNameNotReserved
```
Comment (thaJeztah) [inline container/view.go:291]: FWIW; we probably should think about making `ErrNameNotReserved` and `ErrNameReserved` a proper `errdef` as well (perhaps "conflict" for `ErrNameReserved` ?); I left that for a later exercise for now 😅

**Comment by thaJeztah** [comment]
@rumpl @tianon PTAL 🤗

**Review by tianon — APPROVED** [review]

File: container/view.go [hunk 2]
```diff
@@ -523,22 +492,21 @@ type namesByContainerIDIndexer struct{}
 func (e *namesByContainerIDIndexer) FromObject(obj interface{}) (bool, []byte, error) {
 	n, ok := obj.(nameAssociation)
 	if !ok {
-		return false, nil, fmt.Errorf(`%T does not have type "nameAssociation"`, obj)
+		return false, nil, errdefs.InvalidParameter(errors.Errorf(`%T does not have type "nameAssociation"`, obj))
```
Comment (cpuguy83) [inline container/view.go:495]: Is this error something that the user can do something about or would such an error only be produced by a bug in the caller?

Comment (thaJeztah) [reply container/view.go]: I removed the last commit; need to look more in-depth;

While these errors are "invalid parameter", the cause of the invalid parameter (wrong type) is not due to what the user did.

I'll revisit this one, and have a closer look;

- Either: keep InvalidParameter, but make sure it's wrapped to "system" after that
- Or (this commit); consider it "system" errors from the start


It's slightly difficult to follow where these are called, as it's implementing an interface, and that interface is called from the vendored memdb code, so need to look closer

**Comment by thaJeztah** [comment]
@cpuguy83 @vvoland PTAL (now that I removed that one commit)

**Review by rumpl — APPROVED** [review]
👍

File: daemon/list.go [hunk 3]
```diff
@@ -364,8 +367,7 @@ func (daemon *Daemon) foldFilter(ctx context.Context, view *container.View, conf
 
 func idOrNameFilter(view *container.View, value string) (*container.Snapshot, error) {
 	filter, err := view.Get(value)
-	switch err.(type) {
-	case container.NoSuchContainerError:
+	if err != nil && errdefs.IsNotFound(err) {
```
Comment (thaJeztah) [inline daemon/list.go:370]: I could probably simplify this

```suggestion
	if errdefs.IsNotFound(err) {
```

**Review by cpuguy83 — APPROVED** [review]

**Pull request merged**